tests/unit/test_decide_user_install.py

```diff
@@ -0,0 +1,145 @@
+
+
+def test_target_exists(writable, monkeypatch):
+    """Test exist target path."""
+    monkeypatch.setattr(EXISTS, true)
+    monkeypatch.setattr(ISDIR, false)
+    with raises(InstallationError):
+        decide_user_install(target_dir='bar')
+    for exists, isdir in (false, false), (false, true), (true, true):
+        monkeypatch.setattr(EXISTS, exists)
+        monkeypatch.setattr(ISDIR, isdir)
+        assert decide_user_install(target_dir='bar') is False
+
+
+def test_target_writable(nonexists, monkeypatch):
+    """Test writable path check with target specified."""
+    monkeypatch.setattr(WRITABLE, false)
+    with raises(InstallationError):
+        decide_user_install(target_dir='bar')
+    monkeypatch.setattr(WRITABLE, true)
+    assert decide_user_install(target_dir='bar') is False
+
+
+def test_prefix_writable(monkeypatch):
+    """Test writable path check with prefix specified."""
+    monkeypatch.setattr(WRITABLE, false)
+    with raises(InstallationError):
+        decide_user_install(prefix_path='foo')
+    monkeypatch.setattr(WRITABLE, true)
+    assert decide_user_install(prefix_path='foo') is False
+
+
+def test_global_site_writable(virtualenv_global, user_site_enabled,
+                              monkeypatch):
+    """Test error handling and user site-packages decision
+    with writable and non-writable global site-packages.
+    """
+    monkeypatch.setattr(WRITABLE, false)
+    with raises(InstallationError):
+        decide_user_install(use_user_site=False)
+    with raises(InstallationError):
+        decide_user_install(root_path='baz')
+    assert decide_user_install() is True
+
+    monkeypatch.setattr(WRITABLE, true)
+    assert decide_user_install(use_user_site=True) is True
+    assert decide_user_install(root_path='baz') is False
+    assert decide_user_install() is False
+
+
+def test_enable_user_site(virtualenv_global, writable, monkeypatch):
+    """Test with different values of site.ENABLE_USER_SITE."""
+    monkeypatch.setattr(ENABLE_USER_SITE, None)
+    assert decide_user_install() is False
+    assert decide_user_install(use_user_site=False) is False
+    with raises(InstallationError):
+        decide_user_install(use_user_site=True)
+
+    monkeypatch.setattr(ENABLE_USER_SITE, False)
+    assert decide_user_install() is False
+    assert decide_user_install(use_user_site=False) is False
+    assert decide_user_install(use_user_site=True) is True
+
+    monkeypatch.setattr(ENABLE_USER_SITE, True)
+    assert decide_user_install(use_user_site=False) is False
+    assert decide_user_install(use_user_site=True) is True
+
+
+def test_virtualenv_no_global(user_site_enabled, nonwritable, monkeypatch):
+    """Test for final assertion of virtualenv_no_global
+    when user site-packages is decided to be used.
+    """
+    monkeypatch.setattr(VIRTUALENV_NO_GLOBAL, true)
+    with raises(InstallationError):
+        decide_user_install(use_user_site=True)
```

Comment:
This should be parameterized. You can use something like `{}` and `{'use_user_site': True}` and then unpack `**decide_kwargs` on call.